pkg/kapp/diff/versioned_resource.go

```diff
@@ -91,6 +91,34 @@ func (d VersionedResource) updateAffected(rule ctlconf.TemplateRule, rs []ctlres
 		}
 	}
 
+	for _, res := range rs {
+		for k, v := range res.Annotations() {
+			if k == explicitReferenceKey || strings.HasPrefix(k, explicitReferenceKeyPrefix) {
+				explicitRef := NewExplicitVersionedRef(k, v)
+				objectRef, err := explicitRef.AsObjectRef()
+				if err != nil {
+					return fmt.Errorf("Parsing versioned explicit ref on resource '%s': %s", res.Description(), err)
+				}
+
+				// Passing empty TemplateAffectedObjRef as explicit references do not have a special name key
+				err = d.buildObjRefReplacementFunc(ctlconf.TemplateAffectedObjRef{})(objectRef)
+				if err != nil {
+					return fmt.Errorf("Parsing versioned explicit ref on resource '%s': %s", res.Description(), err)
+				}
+
+				annotationMod, err := explicitRef.AnnotationMod(objectRef)
+				if err != nil {
+					return fmt.Errorf("Parsing versioned explicit ref on resource '%s': %s", res.Description(), err)
```

Comment:
also not parsing...

Reply:
Same reason as above. Alternatively,
```suggestion
					return fmt.Errorf("Updating versioned explicit ref on resource '%s': %s", res.Description(), err)
```
(Cannot think of a better one at the moment)

Reply:
preparing annotation modification or something like that. point being here that if we are going to add some context to the error, it should be unique (otherwise we ll be looking at the wrong piece of code if its all the same), and ideally provides a little bit more info to the end user -- not really useful in this specific case.

Reply:
Caught your drift! Making the changes